content/docs/1.3.0/volumes-and-nodes/volume-size.md

```diff
@@ -91,24 +105,37 @@ Here we summarize the important things related to disk space usage we have in th
   Deleting then writing new files would lead to the actual size keeps increasing. Since the filesystem may not reuse the recently freed blocks from recently deleted files. Thus, allocating an appropriate nominal size for a volume that holds heavy writing tasks according to the IO pattern would make disk space usage more efficient.
 
 - By deleting snapshots, the overlapping part of the used blocks might be eliminated regardless of whether the blocks are recently released blocks by the filesystem or still contain historical data.
+
 ## Space Configuration Suggestions for Volumes
 
 1. Reserve enough free space in disks as buffers in case of the actual size of existing volumes keep growing up.
-    - A quick estimation for maximum space consumption estimation of a volume is
-  
+    - A general estimation for maximum space consumption estimation of a volume is
+
+        ```
+        (N + 1) x head/snapshot average actual size
+        ```
+
+        where `N` is the total number of snapshots the volume contains (including the volume head), and the extra `1` is for the temporary space that may be required by snapshot deletion.
+    - If the snapshotting is handled by one recurring job and the retention number is N. Then the formula is
         ```
-        (N + 1 + 1) x head/snapshot average actual size
+        (M + N + 1 + 1 + 1 + 1) x head/snapshot average actual size
         ```
 
-        where N is the number of snapshots retained, the 1st 1 is for the system snapshot, and the 2nd 1 is for the temporary space that may be required by snapshot deletion.
+        where `M` is the snapshots created by users manually, `N` is the snapshot recurring job retain number, the 1st `1` means the volume head, the 2nd `1` is the system snapshots created by rebuilding or expansion operations, the 3rd `1` is the extra snapshot before the recurring job starts cleanup, and the 4th `1` is for the temporary space that may be required by snapshot deletion.
+   - If users don't want snapshot at all. The formula would be
+       ```
+       (1 + 1 + 1) x head/snapshot average actual size
+       ```
+
+       where the 1st `1` means the volume head, the 2nd `1` is the first system snapshot created by rebuilding or expansion operations, the 3rd `1` is the second system snapshot, and the 4th `1` is for the temporary space that may be required by snapshot deletion/merging the 1st system snapshot with the 2nd one.
```

Comment:
This fomular is bigger than the number that we gave UiPath (3x). Should we mention that this is the absolute worst case? The arrange case should be 3x?

Reply:
Not sure. Typically 3x is large enough. This is actually the corner case:
1. Volume head is full
2. 1st Rebuilding is triggered. (The purge of this rebuilding cannot do anything.)
3. Before there is another pruning coming, the volume head suddenly becomes full and the 2nd rebuilding is triggered.
4. During the purge (for the 2nd rebuilding), the volume head becomes full

Reply:
One question: will the snapshot purging and rebuilding/expanding happen simultaneously?

Reply:
IIRC, No. The snapshot purge is the last step of the rebuilding, which happens after the new replica becomes RW.